src/sdk/bucketspace_fs.js

```diff
@@ -747,9 +756,11 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
         // If the system owner account wants to access the bucket, allow it
         if (is_system_owner) return true;
         const is_owner = (bucket.bucket_owner.unwrap() === account_identifier);
+        const is_iam_and_same_root_account_owner = account.owner !== undefined &&
+            account.owner_name.unwrap() === bucket.bucket_owner.unwrap(); // case for IAM account
```

Comment:
use bucket.owner_account (id) instead of name

Reply:
changed, thanks.